torch/csrc/jit/tensorexpr/cuda_codegen.h

```diff
@@ -96,9 +104,30 @@ class GPUMetaVarRewriter : public IRMutator {
   }
 
  private:
+  // When processing a block, stores the contents of each sub-segment.
+  struct Segment {
```

Comment:
Minor: this feels like a class to me. But it's up to you.

Reply:
This is the kind of distinction I don't particularly care about, but happy to change it to class if you think it's clearer.